fusilli-plugin/include/graph_import.h

```diff
@@ -0,0 +1,247 @@
+  // incoming device buffers (identified by UID) to their corresponding
+  // fusilli::TensorAttr.
+  //
+  // All tensors in this map should be non-virtual boundary tensors---a virtual
+  // tensor is an internal intermediate---internal tensors will be tracked
+  // elsewhere. Currently we only support single node graphs, so there are no
+  // virtual tensors to track.
+  std::unordered_map<int64_t, std::shared_ptr<fusilli::TensorAttr>>
+      uidToIOTensor;
+
+  // Helper class for reading from flatbuffer.
+  hipdnn_plugin::GraphWrapper opGraphWrapper;
+
+  GraphImport(const hipdnnPluginConstData_t *opGraph)
+      : opGraphWrapper(opGraph->ptr, opGraph->size) {}
+
+  fusilli::ErrorObject importGraph() {
+    const hipdnn_sdk::data_objects::Graph &hipDnnGraph =
+        opGraphWrapper.getGraph();
+
+    // Import graph level properties.
+    fusilliGraph.setName(hipDnnGraph.name()->str())
+        .setIODataType(
+            FUSILLI_TRY(convertHipDnnToFusilli(hipDnnGraph.io_type())))
+        .setIntermediateDataType(FUSILLI_TRY(
+            convertHipDnnToFusilli(hipDnnGraph.intermediate_type())))
+        .setComputeDataType(
+            FUSILLI_TRY(convertHipDnnToFusilli(hipDnnGraph.compute_type())));
+
+    return importNodes();
+  }
+
+  // Import all graph nodes.
+  fusilli::ErrorObject importNodes() {
+    if (opGraphWrapper.nodeCount() > 1)
+      return fusilli::error(fusilli::ErrorCode::NotImplemented,
+                            "Multi-node graphs not supported currently.");
+    for (size_t i = 0; i < opGraphWrapper.nodeCount(); ++i) {
+      const hipdnn_sdk::data_objects::Node &node = opGraphWrapper.getNode(i);
+      FUSILLI_CHECK_ERROR(importNode(node));
+    }
+
+    return fusilli::ok();
+  }
+
+  // Import single graph node.
+  fusilli::ErrorObject importNode(const hipdnn_sdk::data_objects::Node &node) {
+    switch (node.attributes_type()) {
+    case hipdnn_sdk::data_objects::NodeAttributes::ConvolutionFwdAttributes:
+      FUSILLI_CHECK_ERROR(
+          importConvFPropAttr(node.attributes_as_ConvolutionFwdAttributes()));
+      break;
+    default:
+      return fusilli::error(fusilli::ErrorCode::NotImplemented,
+                            "Unsupported node type.");
+    }
+    return fusilli::ok();
+  }
+
+  fusilli::ErrorObject
+  importConvFPropAttr(const hipdnn_sdk::data_objects::ConvolutionFwdAttributes
+                          *hipDnnConvFwdAttr) {
+    // Import node inputs.
+    std::shared_ptr<fusilli::TensorAttr> x =
+        FUSILLI_TRY(importNodeInput(hipDnnConvFwdAttr->x_tensor_uid(), "x"));
+    std::shared_ptr<fusilli::TensorAttr> w =
+        FUSILLI_TRY(importNodeInput(hipDnnConvFwdAttr->w_tensor_uid(), "w"));
+
+    // Import node.
+    auto fusilliConvFwdAttr =
+        fusilli::ConvFPropAttr()
+            // TODO(#2369): pre_padding and post_padding vs padding
+            .setPadding(*hipDnnConvFwdAttr->post_padding())
```

Comment:
We could probably assert here that post_padding == pre_padding, since Fusilli only supports `padding` (so symmetric).

Reply:
+1 we should add a check to `hipdnnEnginePluginGetApplicableEngineIds` (and a test for said check) to ensure we never get to the point where we are attempting to create an execution context for a conv node with asymmetric padding as well.
https://github.com/nod-ai/shark-ai/blob/a56c94ba8f9e48df14439955db1b208dc3ce188d/fusilli-plugin/src/fusilli_plugin.cpp#L210-L226